docs/getting-started.md

```diff
@@ -112,45 +140,83 @@ In the application, anyone can vote with the same voting logic, and hence can us
 
 Now you are ready to execute the contract with the following command.
 
-```
-$ client/bin/execute-contract --properties client.properties --contract-id StateUpdater --contract-argument '{"asset_id":"some_asset", "state":3}'
+```shell
+client/bin/execute-contract --properties client.properties --contract-id StateUpdater --contract-argument '{"asset_id":"some_asset", "state":3}'
 ```
 
 In the contract argument, the value specified with the key `asset_id` must be unique globally for each asset.
-In Scalar DL 1.0, `asset_id` is not a reserved json key name and you can use any json key name, but this might be changed in future versions.
+
+## Validate the states of Ledger
+
+You can validate the states of Ledger by executing the following command.
+
+```shell
+client/bin/validate-ledger --properties client.properties --asset-id="some_asset"
+```
+
+What the validation does is depending on how you set up and configure Scalar DL.
+Briefly speaking, with only Scalar DL Ledger, the validation traverses assets to see if the assets can be recomputed and have valid hash-chain structure.
+With Scalar DL Ledger and Auditor, the validation checks discrepancies (i.e., Byzantine faults) between the states of Ledger and Auditor without a centralized coordination.
+Please read [Getting Started with Scalar DL Auditor](getting-started-auditor.md) for more details about the validation with Auditor.
 
 ## Create your own contracts
 
-As we explained above, what you need to create your own contracts is extend the `Contract` class and override the `invoke` method as you like.
-We are preparing more sample contracts, so please wait for an update.
+As we explained above, what you need to create your own contracts is to extend the predefined base contract classes and override the `invoke` method as you like.
+As of writing this, we provide three base contracts:
+* [JacksonBasedContract](https://scalar-labs.github.io/scalardl/javadoc/ledger/com/scalar/dl/ledger/contract/JacksonBasedContract.html)
+* [JsonpBasedContract](https://scalar-labs.github.io/scalardl/javadoc/ledger/com/scalar/dl/ledger/contract/JsonpBasedContract.html)
+* [StringBasedContract](https://scalar-labs.github.io/scalardl/javadoc/ledger/com/scalar/dl/ledger/contract/StringBasedContract.html)
+
+They differ in the way to serialize and deserialize arguments and data that are read and written via Contract and Ledger interfaces.
+JacksonBasedContract uses [Jackson]() to serialize a JSON object (i.e., [JsonNode]()) to internal String representation and deserialize a String object to a [JsonNode]() object. 
+In the same way, JsonpBasedContract uses [JSONP]() for JSON serialization and deserialization.
+StringBasedContract does not do JSON serialization and deserialization and uses internal [String]() representation as it is.
```

Comment:
I think it could be convenient if we provide a matrix that users can quickly check what type of arguments they need to use in the application to match the contract.

|Contract Type|Argument Type|
|Contract (deprecated)|JsonObject|
|JsonpBasedContract|JsonObject|
|JacksonBasedContract (recommended)|JsonNode|
|StringBasedContract|String|

What do you think?

Reply:
It was added but moved to another doc in [aa1740b](https://github.com/scalar-labs/scalardl/pull/26/commits/aa1740bf9667366c0312bb73589fe3bcc8141b6b) and https://github.com/scalar-labs/scalardl/pull/28.
Anyway, thank you!